File: lis.c
#include <stdio.h>
#include <stdlib.h>

int get_lis(int array[], int len);

int main(int argc, char *argv[]){

  int array[] = {0, 22, 9, 33, 21, 50, 41, 60, 80};
  printf("lis is %d\n", get_list(array, sizeof(array)/sizeof(array[0])));
  return 0;
}

int get_list(int array[], int len){

  int i, j;
  int tmp[len];
  int max = 1;

  
  if(len == 1)
    return 1;

  // init the result array
  for(i=0; i<len; i++)
    tmp[i] = 1;
   
  for(i=1; i<len; i++){ //get the number
    for(j=0; j<i; j++){
      if((array[j] < array[i])&&(tmp[i] <tmp[j]+1))
        tmp[i] = tmp[j] + 1;
    }
  }
 
  //get the largest one 
  for(i=1; i<len; i++){
    if(tmp[i] > max)
      max = tmp[i];
  }
  
  return max;
}

File: java/150/lis.java
//Longest Increasing Subsequence
public class lis{


  public static int getlis(int[] a){
    if(a.length == 1)
	return 1;
    int[] tmp = new int[a.length];
    for(int i=0; i<a.length; i++)
	tmp[i] = 1;

   
    for(int i=1; i<a.length; i++){
      for(int j=0; j<i; j++){
	if((a[i] > a[j]) && (tmp[i] < tmp[j] + 1))
	  tmp[i] = tmp[j] + 1;
      }
    } 
    int max = Integer.MIN_VALUE;

    for(int i=0; i<tmp.length; i++){
	if(max < tmp[i])
	   max = tmp[i];
    }
    return max;
  }

  public static void main(String[] args){
   int[] a = {0, 22, 9, 33, 21, 50, 41, 60, 80};
   System.out.printf("%d\n", getlis(a));
  }
}
